### PR TITLE
merging all the crazy work to master

### PR DESCRIPTION
**Pull request by nyurik**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,57 @@
...
+
+Full planet Sophox should be installed on a sufficiently large (40+ GB RAM, 1TB Disk) server, preferably SSD NVMe disk.  In case of Google Cloud, a local SSD scratch disk is also recommended.  Use environment variables to override what data gets loaded - see [docker/startup.local.sh](docker/startup.local.sh) how to run Sophox locally and with a small OSM file.   If ran with `startup.local.sh`, use  http://sophox.localhost  to browse it.
+
+The server must have `bash`, `docker`, `curl`, and `git`.  Everything else is loaded inside docker containers.
+
+When cloning, make sure you get submodules (e.g. `git submodule update --init --recursive`)
+
+### Google Cloud
+* Create a `custom-6-39936` VM (6 vCPUs, 36GB RAM) or better with a 15GB boot disk, and attach a 1TB Persisted SSD disk.
+* Set VM startup script to the following line, and the service should be ready in two to three days.  You may override   Insert any env var overrides right before, e.g. `export SOPHOX_HOST=example.org; curl ... | bash`
```
Comment (nickpeihl) [inline README.md]: > You may override   Insert any env var...

This sentence looks incomplete.

File: docker/docker-compose.yml [hunk 2]
```diff
@@ -1,90 +1,293 @@
-version: '3'
+version: '2.2'
```
Comment (nickpeihl) [inline docker/docker-compose.yml:1]: Any particular reason to use `2.2` rather than `3`?

File: docker/startup.sh [hunk 3]
```diff
@@ -1,77 +1,223 @@
...
+BLAZEGRAPH_JNL_DATA_FILE=/app-data/osmdata.jnl
+
+# If DEBUG env is not set, run docker compose in the detached (service) mode
+DETACH_DOCKER_COMPOSE=$( [[ "${DEBUG}" == "" ]] && echo "true" || echo "" )
+# Do not use https redirect and Let's Encrypt certs when debugging
+TRAEFIK_FILE=$( [[ "${DEBUG}" == "" ]] && echo "${REPO_DIR}/docker/traefik.toml" || echo "${REPO_DIR}/docker/traefik.debug.toml" )
+TRAEFIK_HOST=$( [[ "${DEBUG}" == "" ]] && echo "0.0.0.0" || echo "127.0.0.1" )
+
+# Get total system memory, reducing it by some optional percentage, in MB
+TOTAL_MEMORY_MB=$(( $(free | awk '/^Mem:/{print $2}') * ${TOTAL_MEMORY_PRCNT} / 100 / 1024 ))
```
Comment (nickpeihl) [inline docker/startup.sh]: `free` is not available on macos. https://apple.stackexchange.com/questions/4286/is-there-a-mac-os-x-terminal-version-of-the-free-command-in-linux-systems

**Review by nickpeihl — CHANGES_REQUESTED** [review]

File: docker/docker-compose.yml (hunk 2 above)
Comment (nyurik) [reply docker/docker-compose.yml:1]: I had 3 before, but in 3 they removed [mem_limit](https://docs.docker.com/compose/compose-file/compose-versioning/#version-3). See https://github.com/docker/compose/issues/4513 (hugely down-voted btw).  Docker author advises to stay with 2 :)

File: docker/startup.sh (hunk 3 above)
Comment (nyurik) [reply docker/startup.sh]: i tried a few of them thanks to @itsmed, but alas. I have added a `MAX_MEMORY_MB` -- if you set it, it won't try to auto-guess it.

**Review by nickpeihl — APPROVED** [review]

**Pull request merged**